### PR TITLE
Option for `hookModule`

### PR DESCRIPTION
**Pull request by roberth**
This removes a bit of duplication, and allows `hookModule` to be monkey patched.

### CODE REVIEW

File: modules/hooks.nix [hunk 1]
```diff
@@ -44,6 +38,20 @@ in
...
+      Base module that must be loaded into each hook.
+    '';
+  };
+
+  config.hookModule = {
+    imports = [ ./hook.nix ];
+    config._module.args.default_stages = cfg.default_stages;
+  };
+  config._module.args.hookModule = config.hookModule;
+
```
Comment (roberth) [inline modules/hooks.nix:54]: This chunk could be moved out to a separate file if you like.

**Comment by roberth** [comment]
> `_module.args` are way more powerful than I initially thought.


You can't use it for imports in the same fixpoint (here: the root, or initial `evalModules`), but you can refer to it in submodules. Those are independent `evalModules` calls performed by the submodule types.

**Comment by sandydoo** [comment]
@azahi, fixed

**Comment by SebTM** [comment]
Thanks for the fast fix 🥇 Works fine again on `8ae5bb63f9bcd0472b67cf4ada414ae37f1f2bcc` (latest master) with both options:

```
inherit (self.checks.${system}.pre-commit-check) shellHook;
buildInputs = self.checks.${system}.pre-commit-check.enabledPackages;
```

**Pull request merged**